lib/inMemFileDB.js

```diff
@@ -66,26 +69,18 @@ class InMemoryFileDB {
 
         // path is always relative to appName.js-controller
         this.dataDir = (this.settings.connection.dataDir || getDefaultDataDir());
-        if (this.dataDir) {
-            if (this.dataDir[0] === '.' && this.dataDir[1] === '.') {
-                this.dataDir = __dirname + '/../' + this.dataDir;
-            } else if (this.dataDir[0] === '.' && this.dataDir[1] === '/') {
-                this.dataDir = __dirname + '/../' + this.dataDir.substring(2);
-            }
-        }
-        this.dataDir = path.normalize(this.dataDir);
+        this.dataDir = path.normalize(path.join(__dirname, '..', this.dataDir) + path.sep);
         this.dataDir = this.dataDir.replace(/\\/g, '/');
-        if (this.dataDir[this.dataDir.length - 1] !== '/') this.dataDir += '/';
 
         // Create data directory
         if (!fs.existsSync(this.dataDir)) {
             fs.mkdirSync(this.dataDir);
         }
 
-        this.datasetName = this.dataDir + this.settings.fileDB.fileName;
+        this.datasetName = path.join(this.dataDir, this.settings.fileDB.fileName);
         this.stateTimer = null;
 
-        this.backupDir = this.settings.backup.path || (this.dataDir + this.settings.fileDB.backupDirName + '/');
+        this.backupDir = this.settings.backup.path || (path.join(this.dataDir, this.settings.fileDB.backupDirName) + path.sep);
```

Comment:
Addinig path.sep should not be necessary if we use the path methods everywhere. We should avoid having to manually add slashes at the end of a path - just another source for errors.

Reply:
path.join will not add tailing path separators, but we want tohave one here ... ok I can check again whey/where

Reply:
> path.join will not add tailing path separators, but we want tohave one here

I know but I'm arguing we might/should not need it.

Reply:
changed in next commit